test/SerializerTest.cpp

```diff
@@ -559,3 +564,214 @@ TEST(Serializer, serializeOptional) {
   EXPECT_THAT(sExpected, ::testing::Optional(std::string("hallo")));
   EXPECT_EQ(nilExpected, std::nullopt);
 }
+
+// _____________________________________________________________________________
+// Tests for CompressedSerializer
+// _____________________________________________________________________________
+
+// Simple dummy "compression" for testing (modify the data in a way that is
+// simple, and reversed in the below dummy decompression function)
+auto dummyCompress = [](ql::span<const char> data) {
+  auto res = std::vector<char>(data.begin(), data.end());
+  for (auto& c : res) {
+    ++c;
+  }
+  res.push_back('A');
```

Comment:
Why the final push_back and pop_back?

Reply:
Just for the style (s.t. they don't have the exact same size, and I can have some additional checks). Not meant as a compression at all.